codalab/lib/canonicalize.py

```diff
@@ -115,12 +115,7 @@ def get_target_path(bundle_store, model, target):
     Return the on-disk location of the target (bundle_uuid, subpath) pair.
     """
     (uuid, path) = target
-    bundle = model.get_bundle(uuid)
-    if not bundle.data_hash:
-        # Note that the bundle might not be ready, but return the location anyway to the temporary directory.
-        bundle_root = get_current_location(bundle_store, uuid)
```

Comment:
Currently, while the bundle is running, data_hash = null, and we access the temp directory for the bundle.  This logic is still necessary, I think.

But actually, we could just not use the temporary directory at all and do everything directly in the data directory, including uploads (in which case the status should be 'uploading', analogous to 'running').  This would require a few more changes in other parts of the code.  I think it should be fine.